tests/installation/agama_reboot.pm

```diff
@@ -58,10 +58,22 @@ sub verify_agama_auto_install_done_cmdline {
     die "Install phase is not done, please check agama logs";
 }
 
+sub expect_is_shown {
+    my $timeout = 2400;
+    while ($timeout > 0) {
+        my $ret = check_screen('agama-install-finished', 30);
+        $timeout -= 30;
+        diag("left total await_install timeout: $timeout");
+        last if $ret;
+        die "timeout ($timeout) hit awaiting installation to be finished" if $timeout <= 0;
+    }
+}
+
```

Comment:
For a quick glance, I don't think your need this function to check the installation process.

Reply:
Yup, you can add a new needle https://openqa.suse.de/tests/17287130#step/agama_reboot/2 [it is 66% matched], btw, it is painful for hyper-v and vmware always when matching needles. :(